galasa-ui/src/app/test-runs/[slug]/page.tsx

```diff
@@ -34,31 +32,34 @@ const TestRun = ({ params: { slug } }: TestRunProps) => {
   };
 
   const fetchRunDetailLogs = async () => {
-
     const rasApiClient = new ResultArchiveStoreAPIApi(apiConfig);
     const rasRunLogsResponse = await rasApiClient.getRasRunLog(slug);
-
     return rasRunLogsResponse;
-
   };
 
   const fetchTestArtifacts = async (): Promise<ArtifactIndexEntry[]> => {
-  
     let runArtifacts: ArtifactIndexEntry[] = [];
-  
     const rasApiClient = new ResultArchiveStoreAPIApi(apiConfig);
     const rasArtifactResponse = await rasApiClient.getRasRunArtifactList(slug);
-  
+    
     if (rasArtifactResponse) {
-      runArtifacts = structuredClone(Array.from(rasArtifactResponse)); //Convert the set into array
+      runArtifacts = structuredClone(Array.from(rasArtifactResponse));
     }
-  
+    
     return runArtifacts;
   };
 
+  // Await all the data
+  const runDetails = await fetchRunDetailsFromApiServer();
+  const runArtifacts = await fetchTestArtifacts();
+  const runLogs = await fetchRunDetailLogs();
```

Comment:
No error handling here? If any of these throw errors, then we'll run into issues.